cdap-ui/app/features/hydrator/services/detail/stores/pipeline-detail-nodeconfig-store.js

```diff
@@ -37,7 +37,17 @@ angular.module(PKG.name + '.feature.hydrator')
     this.setState = function(plugin) {
       onPluginChange.call(this, plugin);
     };
+    this.setPlugin = function(plugin) {
+      this.plugin = plugin;
+      var type = this.isSource ? 'source': false;
```

Comment:
:+1: 
I think it makes the javascript code less clean when having individual booleans.